matchmake-extension/auto_matchmake_postpone.go

```diff
@@ -0,0 +1,102 @@
+		logger.Warning("MatchmakeExtension::AutoMatchmake_Postpone missing CleanupSearchMatchmakeSessionHandler!")
+		missingHandler = true
+	}
+	if missingHandler {
+		return
+	}
+	//This is the best way I found to copy the full object. And I still hate it.
+	tmp, _ := json.Marshal(matchmakeSession)
+	matchmakeSessionCopy := match_making.NewMatchmakeSession()
+	json.Unmarshal(tmp, &matchmakeSessionCopy)
+	searchMatchmakeSession := commonMatchmakeExtensionProtocol.CleanupSearchMatchmakeSessionHandler(*matchmakeSessionCopy)
+	sessionIndex := FindSearchMatchmakeSession(searchMatchmakeSession)
+	if sessionIndex == math.MaxUint32 {
+		session := common_globals.CommonMatchmakeSession{
+			SearchMatchmakeSession: searchMatchmakeSession,
+			GameMatchmakeSession:   *matchmakeSession,
+		}
+		sessionIndex = len(common_globals.Sessions)
+		common_globals.Sessions = append(common_globals.Sessions, session)
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.ID = uint32(sessionIndex)
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.OwnerPID = client.PID()
+		common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.HostPID = client.PID()
+	}
+
+	common_globals.Sessions[sessionIndex].PlayersByConnectionId = append(common_globals.Sessions[sessionIndex].PlayersByConnectionId, client.ConnectionID())
+
+	rmcResponseStream := nex.NewStreamOut(globals.NEXServer)
+	rmcResponseStream.WriteString("MatchmakeSession")
+	lengthStream := nex.NewStreamOut(globals.NEXServer)
+	lengthStream.WriteStructure(&common_globals.Sessions[sessionIndex].GameMatchmakeSession)
+	matchmakeSessionLength := uint32(len(lengthStream.Bytes()))
+	rmcResponseStream.WriteUInt32LE(matchmakeSessionLength + 4)
+	rmcResponseStream.WriteUInt32LE(matchmakeSessionLength)
+	rmcResponseStream.WriteStructure(&common_globals.Sessions[sessionIndex].GameMatchmakeSession)
+
+	rmcResponseBody := rmcResponseStream.Bytes()
+
+	rmcResponse := nex.NewRMCResponse(matchmake_extension.ProtocolID, callID)
+	rmcResponse.SetSuccess(matchmake_extension.MethodAutoMatchmake_Postpone, rmcResponseBody)
+
+	rmcResponseBytes := rmcResponse.Bytes()
+
+	responsePacket, _ := nex.NewPacketV0(client, nil)
+
+	responsePacket.SetVersion(0)
+	responsePacket.SetSource(0xA1)
+	responsePacket.SetDestination(0xAF)
+	responsePacket.SetType(nex.DataPacket)
+	responsePacket.SetPayload(rmcResponseBytes)
+
+	responsePacket.AddFlag(nex.FlagNeedsAck)
+	responsePacket.AddFlag(nex.FlagReliable)
+
+	globals.NEXServer.Send(responsePacket)
+
+	rmcMessage := nex.NewRMCRequest()
+	rmcMessage.SetProtocolID(notifications.ProtocolID)
+	rmcMessage.SetCallID(0xffff0000 + callID)
+	rmcMessage.SetMethodID(notifications.MethodProcessNotificationEvent)
+
+	oEvent := notifications.NewNotificationEvent()
+	oEvent.PIDSource = common_globals.Sessions[sessionIndex].GameMatchmakeSession.Gathering.HostPID
+	oEvent.Type = 3001 // New participant
```

Comment:
Should we break these things out into something more readable? It would be nice to not have to rely on previous knowledge or other peoples comments to know what certain types and flags and such are in some structures like here

Not sure where that should go, though. Maybe as part of the notifications/nintendo-notifications protocols in nex-protocols-go? That way they could be accessed both here and in servers which might want to send notifications manually. Something like `notifications.NotificationTypes.NewMatchmakeParticipant` would be nice